controllers/metal3.io/hostfirmwaresettings_controller.go

```diff
@@ -228,34 +235,46 @@ func (r *HostFirmwareSettingsReconciler) updateStatus(info *rInfo, settings meta
 	generation := info.hfs.GetGeneration()
 
 	if specMismatch {
-		setCondition(generation, &info.hfs.Status, metal3v1alpha1.FirmwareSettingsChangeDetected, metav1.ConditionTrue, reason, "")
+		if setCondition(generation, &newStatus, info, metal3v1alpha1.FirmwareSettingsChangeDetected, metav1.ConditionTrue, reason, "") {
```

Comment:
Don't we need to copy over the existing condtions to the newStatus first? Otherwise the state will always appear to be transitioning.

Reply:
No I don't believe so, setCondition() now updates the condition in newStatus and checks whether its different than the current one to set the dirty flag.  Status().Update will only be called if the Conditions have changed.